src/hooks/useGetBatches.ts

```diff
@@ -0,0 +1,244 @@
+  tokens_out: { symbol: string; address: string; name: string }[]
+  num_trades: number
+  total_value: number
+}
+
+type SubgraphSettlementsResponse = SettlementsResponse
+
+export type Batch = SettlementsResponse
+export type Trade = {
+  id: string
+  timestamp: number
+  gasPrice: BigInt
+  feeAmount: BigInt
+  txHash: string
+  buyAmount: number
+  sellAmount: number
+  sellToken: Token
+  buyToken: Token
+}
+
+type Token = {
+  id: string
+  name: string
+  symbol: string
+  address: string
+  decimals: number
+  priceUsd: number
+  totalVolumeUsd: string
+  hourlyTotals: TokenHourlyTotals[]
+}
+type User = {
+  id: string
+  address: string
+  firstTradeTimestamp: number
+  isSolver: boolean
+  numberOfTrades: number
+}
+
+export type TokenDailyTotalsResponse = {
+  timestamp: number
+  totalVolumeUsd: string
+  token: TokenResponse
+}
+export type TokenResponse = {
+  id: string
+  name: string
+  symbol: string
+  address: string
+  decimals: number
+  priceUsd: number
+  totalVolumeUsd: string
+  hourlyTotals: TokenHourlyTotals[]
+}
+
+export type TokenHourlyTotals = {
+  token: { address: string }
+  timestamp: number
+  totalVolumeUsd: string
+  averagePrice: string
+}
+
+export type SubgraphHistoricalDataResponse = {
+  tokenHourlyTotals: Array<TokenHourlyTotals>
+}
+
+export type TokenData = {
+  timestamp: number
+  lastDayUsdVolume?: number
+  lastDayPricePercentageDifference?: number
+  lastWeekPricePercentageDifference?: number
+  lastWeekUsdPrices?: Array<{ time: UTCTimestamp; value: number }>
+}
+
+export type BatchValue = {
+  batchId: string
+  totalValue: number
+}
+const extractBuyTokens = (trades: Trade[]): { name: string; symbol: string; address: string }[] => {
+  const tokens = trades.map((trade) => {
+    const symbol =
+      trade.buyToken.address.toLowerCase() === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+        ? 'ETH'
+        : trade.buyToken.symbol
+    const name =
+      trade.buyToken.address.toLowerCase() === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+        ? 'Ethereum'
+        : trade.buyToken.name
+
+    return { name, symbol, address: trade.buyToken.address }
+  })
+  // Remove duplicates
+  return tokens.filter((token, index) => tokens.findIndex((t) => t.address === token.address) === index)
+}
+
+const extractSellTokens = (trades: Trade[]): { name: string; symbol: string; address: string }[] => {
+  const tokens = trades.map((trade) => {
+    const symbol =
+      trade.sellToken.address.toLowerCase() === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+        ? 'ETH'
+        : trade.sellToken.symbol
+    const name =
+      trade.sellToken.address.toLowerCase() === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+        ? 'Ethereum'
+        : trade.sellToken.name
+
+    return { name, symbol, address: trade.sellToken.address }
+  })
+  // Remove duplicates
+  return tokens.filter((token, index) => tokens.findIndex((t) => t.address === token.address) === index)
+}
```

Comment:
i didn't review fully this file. Needs to be cleaned up. Too big and hard to review.

Reply:
Also, i believe thegraph types should be auto-generated. 

Actually, this should already be part of the SDK (the autogeneration of types)